### ISSUE TITLE
Shell.Current.Navigation.PushAsync Null Reference Exception

### ISSUE THREAD

**Issue by jmdee** [dotnet/maui#9647]
### Description

Hi guys, so I am experiencing a null reference exception each time I call Shell.Current.Navigation.PushAsync(page). 

System.NullReferenceException: 'Object reference not set to an instance of an object.'

Microsoft.Maui.Controls.Shell.Current.get returned null.

![image](https://user-images.githubusercontent.com/18168415/186653891-b4c77956-7f88-4f04-865b-971dd203ac99.png)

### Steps to Reproduce

1) Clone Project from the link below
https://github.com/jmdee/errormauishell 
2) Restore all packages 
3) Debug on Android 11 
4) Once the application is launched, click the button labeled "login" 
5) Click the Click Me Button on the new page. This button will throw the exception

see example code that is throwing the exception 

![image](https://user-images.githubusercontent.com/18168415/186650110-117e9223-5084-4c44-b9c6-44cbb76b3122.png)
error 
{System.NullReferenceException: Object reference not set to an instance of an object.
   at ProjectName.ViewModels.Projects.MyCompaniesPageViewModel.AddCompany() in ....}

### Version with bug

6.0.400

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows

### Affected platform versions

windows10.0.19041.0 , Android 11

### Did you find any workaround?

No

**Comment by PureWeen** [comment]
Can you also include more of the exception so we can see the stack trace?

**Comment by jmdee** [comment]
> ### Description
> Hi guys, so I am experiencing a null reference exception each time I call Shell.Current.Navigation.PushAsync(page).
> 
> System.NullReferenceException: 'Object reference not set to an instance of an object.'
> 
> Microsoft.Maui.Controls.Shell.Current.get returned null.
> 
> ![image](https://user-images.githubusercontent.com/18168415/186653891-b4c77956-7f88-4f04-865b-971dd203ac99.png)
> 
> ### Steps to Reproduce
> 1. Create a New .net maui application.
> 
> * Create MyPageView and MyPage2View
> 
> 2. Create  MyPgeViewmodel and bind it to MyPageView
> 3. Create a RelayCommand in the MyPageViewModel
> 4. Create a Button in MyPageView and add a command binding to the relay command on 3 above
> 5. Inside the relay command and Shell.Current.Navigation.PushAsync(new MyPage2View())
> 
> see example code that is throwing the exception
> 
> ![image](https://user-images.githubusercontent.com/18168415/186650110-117e9223-5084-4c44-b9c6-44cbb76b3122.png) error {System.NullReferenceException: Object reference not set to an instance of an object. at ProjectName.ViewModels.Projects.MyCompaniesPageViewModel.AddCompany() in ....}
> 
> ### Version with bug
> 6.0.400
> 
> ### Last version that worked well
> Unknown/Other
> 
> ### Affected platforms
> Android, Windows
> 
> ### Affected platform versions
> windows10.0.19041.0 , Android 11
> 
> ### Did you find any workaround?
> No
> 
> ### Relevant log output
> _No response_



> Hi @jmdee. We have added the "s/needs-repro" label to this issue, which indicates that we require steps and sample code to reproduce the issue before we can take further action. Please try to create a minimal sample project/solution or code samples which reproduce the issue, ideally as a GitHub repo that we can clone. See more details about creating repros here: https://github.com/dotnet/maui/blob/main/.github/repro.md
> 
> This issue will be closed automatically in 7 days if we do not hear back from you by then - please feel free to re-open it if you come back to this issue after that time.

### Steps to Reproduce

1) Clone Project from the link below
https://github.com/jmdee/errormauishell 
2) Restore all packages 
3) Debug on Android 11 
4) Once the application is launched, click the button labeled "login" 
5) Click the Click Me Button on the new page. This button will throw the exception

**Comment by jmdee** [comment]
> Can you also include more of the exception so we can see the stack trace?

Hi @PureWeen  

### Steps to Reproduce

1) Clone Project from the link below
https://github.com/jmdee/errormauishell 
2) Restore all packages 
3) Debug on Android 11 
4) Once the application is launched, click the button labeled "login" 
5) Click the Click Me Button on the new page. This button will throw the exception

**Comment by jmdee** [comment]
> Can you also include more of the exception so we can see the stack trace?
System.NullReferenceException
  Message=Object reference not set to an instance of an object.

 	0xFFFFFFFFFFFFFFFF in Android.Runtime.JNIEnv.monodroid_debugger_unhandled_exception	C#
 	0x1A in Android.Runtime.JNINativeWrapper._unhandled_exception at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:12,5	C#
 	0x1D in Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:23,26	C#
 	0x17 in System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw	C#
 	0x6 in System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0	C#
 	0xC in Android.App.SyncContext. at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.App/SyncContext.cs:36,19	C#
 	0xE in Java.Lang.Thread.RunnableImplementor.Run at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Thread.cs:36,6	C#
 	0x8 in Java.Lang.IRunnableInvoker.n_Run at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net6.0/android-31/mcw/Java.Lang.IRunnable.cs:84,4	C#
 	0x8 in Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:22,5	C#

**Comment by PureWeen** [comment]
@jmdee that looks like an exception related to an async task that's running and isn't caught

It might also be caused by switching of the `MainPage` 

https://github.com/jmdee/errormauishell/blob/master/Views/LoginPage.xaml.cs#L12

There are currently a few bugs logged around this that we hope to get to

**Comment by jmdee** [comment]
> @jmdee that looks like an exception related to an async task that's running and isn't caught
> 
> It might also be caused by switching of the `MainPage`
> 
> https://github.com/jmdee/errormauishell/blob/master/Views/LoginPage.xaml.cs#L12
> 
> There are currently a few bugs logged around this that we hope to get to

@PureWeen  , is there a current work around because we actually working on a project and this has prevented any progress?

**Comment by jmdee** [comment]
@PureWeen  so turns out it's switching the MainPage, When I set the MainPage to AppShell in the App.xaml.cs the eror goes away. See below. So now what can be a quick win for us to at least continue with the project ?
![image](https://user-images.githubusercontent.com/18168415/186756380-97a78777-65ac-4059-8860-1e1f20a3403a.png)

**Comment by PureWeen** [comment]
@jmdee don't switch between shell and navpage
If you want to use shell just use shell
You shouldn't really need a navpage

You can use "Isvisible" on shell items or remove/add shell items when the user logs in/out

**Comment by PureWeen** [comment]
Duplicate of #10573
